dm/ctl/ctl.go

```diff
@@ -74,9 +66,57 @@ func Start(args []string) {
 		master.NewPurgeRelayCmd(),
 		master.NewMigrateRelayCmd(),
 	)
+}
 
-	rootCmd.SetArgs(args)
+// Init initializes dm-control
+func Init(cfg *common.Config) error {
+	// set the log level temporarily
+	log.SetLevel(zapcore.InfoLevel)
+
+	return errors.Trace(common.InitUtils(cfg))
+}
+
+// PrintUsage prints usage
+func PrintUsage() {
+	maxCmdLen := 0
+	for _, cmd := range rootCmd.Commands() {
+		if maxCmdLen < len(cmd.Name()) {
+			maxCmdLen = len(cmd.Name())
+		}
+	}
+	fmt.Println("Available Commands:")
+	for _, cmd := range rootCmd.Commands() {
+		format := fmt.Sprintf("  %%-%ds\t%%s\n", maxCmdLen)
+		fmt.Printf(format, cmd.Name(), cmd.Use)
+	}
+}
 
+// HasCommand represent whether rootCmd has this command
+func HasCommand(name string) bool {
+	for _, cmd := range rootCmd.Commands() {
+		if name == cmd.Name() {
+			return true
+		}
+	}
+	return false
+}
+
+// PrintHelp print help message for special subCommand
+func PrintHelp(args []string) {
+	cmd, _, err := rootCmd.Find(args)
+	if err != nil {
+		fmt.Println(err)
+		rootCmd.SetOut(os.Stdout)
+		rootCmd.Usage()
+		return
```

Comment:
seems can delete these three lines

Reply:
errors occurred, output all commands' usage is also fine?

Reply:
if error occurred, user may need some help message